data-collection/deploy/module-cost-optimization-hub.yaml

```diff
@@ -115,57 +124,114 @@ Resources:
           sys.path.insert(0,'/tmp/')
 
           import boto3 #pylint: disable=wrong-import-position
+          from botocore.config import Config
 
           logger = logging.getLogger()
+          # Create a new handler here because downloading the updated boto3 affects the
+          # default Lambda logging handler such that only Warning and Error are logged.
+          for h in logger.handlers:
+            logger.removeHandler(h)
+          h = logging.StreamHandler(sys.stdout)
+          h.setFormatter(logging.Formatter('[%(levelname)s] %(message)s'))
+          logger.addHandler(h)
           logger.setLevel(getattr(logging, os.environ.get('LOG_LEVEL', 'INFO').upper(), logging.INFO))
 
           BUCKET_NAME = os.environ['BUCKET_NAME']
           ROLENAME = os.environ['ROLENAME']
           PREFIX = os.environ['PREFIX']
+          DETAIL_SM_ARN = os.environ['DETAIL_SM_ARN']
           TMP_FILE = "/tmp/data.json"
-          NEED_DETAILED = True # set it to False to fallback to collection of only summaries if any perf issue or timeout
```

Comment:
can we still keep option not to pull details?

Reply:
Absolutely. Good idea for some use cases. I added it as configurable via the os param